aws/cloudformation.yml

```diff
@@ -0,0 +1,699 @@
+                usermod -a -G sudo mastodon
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "change mastodon password" >> /var/tmp/bootstrap.txt
+                echo "mastodon:${MastodonLocalUserPassword}" | sudo chpasswd
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Install Docker" >> /var/tmp/bootstrap.txt
+                apt --yes install apt-transport-https ca-certificates curl software-properties-common
+                curl -fsSL https://download.docker.com/linux/ubuntu/gpg | apt-key add -
+                add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"
+                apt update
+                apt-cache policy docker-ce
+                apt --yes install docker-ce
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "grant mastodon user docker" >> /var/tmp/bootstrap.txt
+                usermod -aG docker mastodon
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "install docker-compose" >> /var/tmp/bootstrap.txt
+                curl -L https://github.com/docker/compose/releases/download/1.22.0/docker-compose-$(uname -s)-$(uname -m) -o /usr/local/bin/docker-compose
+                chmod +x /usr/local/bin/docker-compose
+                echo "$?" >> /var/tmp/bootstrap.txt
+
+                echo "Activate Swap" >> /var/tmp/bootstrap.txt
+                fallocate -l 5G /swapfile
+                chmod 600 /swapfile
+                mkswap /swapfile
+                swapon /swapfile
+                echo '/swapfile       none    swap    sw      0       0' | tee -a /etc/fstab
+
+              mode: "000777"
+              owner: "ubuntu"
+              group: "ubuntu"
+
+            /var/tmp/stage2.sh:
+              content: !Sub 
+                - |
+                  #!/bin/bash -ex
+
+                  cd /home/mastodon
+                  echo "git clone" >> /var/tmp/bootstrap.txt
+                  git clone https://github.com/tootsuite/mastodon.git live
+                  echo "$?" >> /var/tmp/bootstrap.txt
+                  cd /home/mastodon/live
+                  echo "git checkout" >> /var/tmp/bootstrap.txt
+                  git checkout ${ReleaseVersion}
+                  echo "$?" >> /var/tmp/bootstrap.txt
+
+                  echo "copy AWS docker-compose.yml" >> /var/tmp/bootstrap.txt
+                  rm -f docker-compose.yml
+                  cp /var/tmp/docker-compose.yml /home/mastodon/live/docker-compose.yml
+
+                  echo "create .env.production" >> /var/tmp/bootstrap.txt
+                  cat >> /home/mastodon/live/.env.production << EOF
+                  LOCAL_DOMAIN=${Domain}
+                  SINGLE_USER_MODE=false
+                  SECRET_KEY_BASE=fc6379db2af899b3c980c83fc725f06996e9c5994b8adf34ec3c346731b6964c7b850209e0edfbc594d26a95108b6fe344bd0c9d2deeee127b51184d75a90cce
```

Comment:
All these secrets need to be parameterized in the YAML, with sensible defaults to highlight the need to change them. We don't want to encourage secret re-use!

Reply:
ok.. this was one of those chicken and egg things....

`docker-compose run --rm web rake secret`
`docker-compose run --rm web rake mastodon:webpush:generate_vapid_key`

requires the docker-build build has completed... but I was generating the .env.production in a previous step.

@Gargron Is it safe to change the secrets AFTER you've done the db:setup? I think they're used for MFA stuff?

Reply:
I believe there is a secrets rotation tool in 2.5.0 with toot-ctl, but I could be mistaken as I haven't seen it documented yet